worker/provisioner/provisioner_task_test.go

```diff
@@ -330,6 +330,32 @@ func (s *ProvisionerTaskSuite) TestZoneConstraintsWithDistributionGroup(c *gc.C)
 	workertest.CleanKill(c, task)
 }
 
+func (s *ProvisionerTaskSuite) TestPopulateAZMachinesErrorWorkerStopped(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	broker := mocks.NewMockZonedEnviron(ctrl)
+	broker.EXPECT().AllRunningInstances(s.callCtx).Return(nil, errors.New("boom"))
+
+	task := s.newProvisionerTaskWithBroker(c, broker, map[names.MachineTag][]string{
+		names.NewMachineTag("0"): {"az1"},
+	})
+
+	res := make(chan error)
+	go func() {
+		res <- task.Wait()
```

Comment:
`DirtyKill` always makes `Wait` return right?

Reply:
It's the same as `CleanKill` (will time-out with an error if wait does not return), but ignores the error.
Actually this means it can be done in simpler fashion; stand by.